spring-kafka/src/main/java/org/springframework/kafka/config/StreamsBuilderFactoryBean.java

```diff
@@ -290,7 +291,12 @@ public synchronized void start() {
 			try {
 				Assert.state(this.streamsConfig != null || this.properties != null,
 						"'streamsConfig' or streams configuration properties must not be null");
-				Topology topology = getObject().build(); // NOSONAR
+				Properties topologyProps = this.properties;
+				if (topologyProps == null) {
+					topologyProps = new Properties();
+					topologyProps.putAll(this.streamsConfig.originals());
```

Comment:
We should probably just dump this deprecated stuff in 2.3; we're now getting deprecation warnings in `start()` with the 2.3.0 client.

Reply:
Well, we need to deprecate a public API, but deal just with the `Properties` internally. Fortunately this `this.streamsConfig.originals()` may help us in setters.

Reply:
We already deprecated it in 2.2.

Reply:
Doh. Missed that. So, yes: we are good to remove `StreamsConfig` support in `2.3`.

Reply:
https://github.com/spring-projects/spring-kafka/issues/1159